packages/blocks/github-pr-overview/src/app.tsx

```diff
@@ -1,241 +1,296 @@
+  const blockRef = React.useRef<HTMLDivElement>(null);
+  const { graphService } = useGraphBlockService(blockRef);
+  const {
+    entityId,
+    // selectedPullRequest is just an Identifier, but isn't the associated GithubPullRequestEntity
+    properties: { selectedPullRequest: remoteSelectedPullRequestId },
+  } = blockEntity;
+  const [
+    {
+      blockState,
+      selectedPullRequestId,
+      allPrs,
+      pullRequest,
+      reviews,
+      events,
+      githubEntityTypeIds,
+      infoMessage,
+    },
+    dispatch,
+  ] = React.useReducer<React.Reducer<LocalState, Actions>>(
+    reducer,
+    getInitialState({
+      selectedPullRequestId: remoteSelectedPullRequestId,
+    }),
+  );
 
   const setSelectedPullRequestIdAndPersist = (
     pullRequestId?: PullRequestIdentifier,
   ) => {
-    if (updateEntities) {
-      updateEntities([
-        {
-          entityId,
-          accountId,
-          data: {
-            selectedPullRequest: pullRequestId,
-          },
+    void graphService?.updateEntity({
+      data: {
+        entityId,
+        properties: {
+          selectedPullRequest: pullRequestId,
         },
-      ]).catch((err) => {
-        throw err;
-      });
-    }
-    setSelectedPullRequestId(pullRequestId);
-  };
+      },
+    });
 
-  const [githubEntityTypeIds, setGithubEntityTypeIds] =
-    React.useState<{ [key in GITHUB_ENTITY_TYPES]: string }>();
+    dispatch({
+      type: "UPDATE_STATE",
+      payload: { selectedPullRequestId: pullRequestId },
+    });
+  };
 
-  const [allPrs, setAllPrs] = React.useState<Map<string, GithubPullRequest>>();
-  const [pullRequest, setPullRequest] = React.useState<GithubPullRequest>();
-  const [reviews, setReviews] = React.useState<GithubReview[]>();
-  const [events, setEvents] = React.useState<GithubIssueEvent[]>();
+  const resetPRInfo = () => {
+    void graphService?.updateEntity({
+      data: {
+        entityId,
+        properties: {
+          selectedPullRequest: undefined,
+        },
+      },
+    });
 
-  /** @todo - Figure out when to query for more than one page, probably querying until no more results */
+    dispatch({ type: "RESET_SELECTED_PR" });
+  };
 
-  React.useEffect(() => {
-    if (isDefined(aggregateEntityTypes) && githubEntityTypeIds === undefined) {
-      setBlockState(BlockState.Loading);
-      getGithubEntityTypes(
-        aggregateEntityTypes,
-        accountId,
-        5,
-        setGithubEntityTypeIds,
-        setBlockState,
-      );
-    }
-  }, [
-    aggregateEntityTypes,
-    githubEntityTypeIds,
-    accountId,
-    setGithubEntityTypeIds,
-    setBlockState,
-  ]);
-
-  // Block hasn't been initialized with a selected Pull Request, get all PRs to allow user to pick
+  // Update local data if remote data changes
   React.useEffect(() => {
     if (
-      selectedPullRequestId === undefined &&
-      githubEntityTypeIds !== undefined
+      JSON.stringify(remoteSelectedPullRequestId) !==
```

Comment:
Why is this necessary? Did 0.2 get implemented with the Ids being objects?

Reply:
This is an object. I renamed `selectedPullRequest` (which is an object) to `selectedPullRequestId` because I felt the name conflicts with the `pullRequest` property we have in the local state. 

https://github.com/hashintel/hash/blob/90859ac19b414dda9db476e6b5e2857e25bbeb6c/packages/blocks/github-pr-overview/src/app.tsx#L84-L85

https://github.com/hashintel/hash/blob/90859ac19b414dda9db476e6b5e2857e25bbeb6c/packages/blocks/github-pr-overview/src/app.tsx#L87-L100

Thinking about it, I guess `selectedPullRequestId` can be assumed to be a string. Happy to revert back to `selectedPullRequest` if you prefer that

Reply:
Gotcha, I was looking at this through GitHub so I wasn't benefiting from intellisense so I didn't easily see the type. I understand the rename, I think the issue I see is that we have actual PullRequest entities, and this is just an identifier for one. So perhaps we can call it `selectedPullRequestIdentifier` or something? Maybe it's just that `id` tends to have connotations of Strings.

I'm not too bothered, though, also more than happy to leave as is because when using an IDE this would be a lot clearer.

Reply:
Updated here => https://github.com/hashintel/hash/pull/705/commits/ac1ebefdd4cd146a393b31476354710c2a52c72e#diff-823b1aa85546193960a17d9c44f706004c4594ceb1214dfd64b48b4b3d22ddf7